### PR TITLE
Task-4

### PR DESCRIPTION
**Pull request by TGreyW**
Task 4.1 is implemented
TASK 4.2 is implemented lambda links are provided and returns data
TASK 4.3 is implemented lambda links are provided and products is stored in DB
My Frontend application is integrated with product service https://d22a5k01kuxglm.cloudfront.net/

POST/products lambda functions returns error 400 BUT ONLY IF NO DATA SPECIFIED - {} Didnt make per field validation
All lambdas return error 500 status code on any error - can be checked providing wrong UUID format
All lambdas do console.log for each incoming requests
Transaction based creation of product

You can playAPI with swagger file: https://editor.swagger.io/
File in commit

### CODE REVIEW

File: productService/data.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+INSERT INTO public.products (id,title,description,price)
+	VALUES ('339465d0-2077-11eb-bbe3-028d774dc6a1','Пирожок с картохой','Сойдет по вкусу',10);
```
Comment (maksym-lebedynets) [inline productService/data.sql:16]: You should not generate IDs on your own. It is the responsibility of SQL

File: productService/handlers/createProduct.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    'Access-Control-Allow-Origin': '*',
+                },
+                body: JSON.stringify({}),
+            };
+        }
+
+        await client.connect();
+        await client.query('BEGIN');
+        await client.query(`
+            INSERT INTO public.products (id,title,description,price)
```
Comment (maksym-lebedynets) [inline productService/handlers/createProduct.js:38]: The same comment about ID

File: productService/handlers/getProductById.js [hunk 3]
```diff
@@ -1,31 +1,72 @@
...
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {
+        rejectUnauthorized: false,
+    },
+    connectionTimeoutMillis: 5000,
+};
```
Comment (maksym-lebedynets) [inline productService/handlers/getProductById.js:17]: What about reuse repeated code?

File: productService/index.js [hunk 4]
```diff
@@ -1,2 +1,3 @@
 module.exports.getProductList = require('./handlers/getProductList');
 module.exports.getProductById = require('./handlers/getProductById');
+module.exports.createProduct = require('./handlers/createProduct');
```
Comment (maksym-lebedynets) [inline productService/index.js:3]: Why do you need this file?

File: productService/handler.js [hunk 5]
```diff
@@ -1 +1,119 @@
...
\ No newline at end of file
+/*
```
Comment (maksym-lebedynets) [inline productService/handler.js:1]: What is this?

**Review by maksym-lebedynets — CHANGES_REQUESTED** [review]

File: productService/handler.js (hunk 5 above)
Comment (lithiumfallen) [reply productService/handler.js:1]: @TGreyW this file should be in raw state without compilation

File: productService/index.js (hunk 4 above)
Comment (lithiumfallen) [reply productService/index.js:3]: @TGreyW in `serverless.yml` functions configured from `handler.createProduct` It's mean that function imported from `handle.js` or `handler/index.js`. This file is redundant as i'm correct. Moreover using webpack you can use `es6` modules

**Comment by lithiumfallen** [comment]
@TGreyW 
Is't work for me. I catch 500 error each time. Could you please check?

**Comment by andreikasmykou** [comment]
Products API doesn't work. Therefore UI app doesn't work too
https://d22a5k01kuxglm.cloudfront.net/
https://9aj51mlxlc.execute-api.eu-west-1.amazonaws.com/dev/products
Can't mark task4

File: productService/handlers/getProductById.js [hunk 6]
```diff
@@ -1,31 +1,72 @@
...
+    const client = new Client(dbOptions);
+    try {
+        await client.connect();
+
+        const { id } = event.queryStringParameters || '';
+        console.log(`getProductById request: id ${id}`);
+        const result = await client.query(
+            `
+                select * from products p 
+                  inner join stock s on s.product_id = p.id
```
Comment (alexeyMohnatkin) [inline productService/handlers/getProductById.js:33]: if stock for this product is missing it will return 0 rows

**Comment by alexeyMohnatkin** [comment]
https://9aj51mlxlc.execute-api.eu-west-1.amazonaws.com/dev/products responds with `500`

**Comment by TGreyW** [comment]
@alexeyMohnatkin 
@lithiumfallen 
@Raerka 

Sorry guys, dont know why, but DB stopped and API GW started throwing 500. Fixed it.
About code, please dont consider code during review. I dont have alot of time, to write "clean and beatiful" code. I do it only for getting experience with AWS, so for now i dont care about my code :) Please fix your scores according to implemented features from our task. And, of course you can subtract some points because of bad code, if you wish. Up to you. 
Thanks :)

**Pull request merged**